test/test-pages/breitbart/source.html

```diff
@@ -2,17 +2,12 @@
 <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" dir="ltr" prefix="og: http://ogp.me/ns# fb: http://ogp.me/ns/fb# dc: http://purl.org/dc/elements/1.1/# v: http://rdf.data-vocabulary.org/# op: http://media.facebook.com/op#" lang="en">
 
 <head>
-    <script type="text/javascript" src="//m.addthis.com/live/red_lojson/300lo.json?si=5887328e015c3caf&amp;bkl=0&amp;bl=1&amp;sid=5887328e015c3caf&amp;pub=ra-536db77a775cf072&amp;rev=v7.9.5-wp&amp;ln=en&amp;pc=men&amp;cb=0&amp;ab=-&amp;dp=www.breitbart.com&amp;fp=tech%2F2016%2F12%2F22%2Fneutral-snopes-fact-checker-david-emery-un-angry-trump-supporters%2F&amp;fr=&amp;of=0&amp;pd=0&amp;irt=0&amp;vcl=0&amp;md=0&amp;ct=0&amp;tct=0&amp;abt=0&amp;cdn=0&amp;pi=1&amp;rb=0&amp;gen=100&amp;chr=UTF-8&amp;mk=Donald%20Trump%2Cfacebook%2CFact%20Check%2Cfake%20news%2CSnopes%2CTech%2Csnopes&amp;colc=1485255313842&amp;jsl=13505&amp;uvs=5887328e084443ac000&amp;skipb=1&amp;callback=addthis.cbs.oln9_77519642505367390">
-        <script type="text/javascript" id="cadmpjs" async="" src="//c1.rfihub.net/js/smarttag.js"></script>
-        <script type="text/javascript" id="pubnationjs" async="" src="//report-ads-to.pubnation.com/dist/pnr.js?t=pn-52225fd0c8484f06"></script>
-    </script>
```

Comment:
@gijsk ,

The XHTML content generated from [`@mozilla.org/xmlextras/xmlserializer;1"`][xmlserializer] contains some invalid `<script>` nodes which our `JSDOMParser` module cannot handle with, like above I removed in the patch.

After removing the below nested `<script>` nodes
```js
<script>
  <script></script>
  <script></script>
</sciprt>
```
then the tests can be passed.

Since `xmlserializer` generated the nested `<script>` nodes which is invalid structure and cannot be passed with [w3c xhtml validation][w3c-xhtml-validation], I think we should fix the issue in `xmlserializer` module not in `JSDOMParser` module, right? What do you think?

[w3c-xhtml-validation]: https://validator.w3.org
[xmlserializer]: http://searchfox.org/mozilla-central/source/toolkit/components/reader/ReaderMode.jsm#442